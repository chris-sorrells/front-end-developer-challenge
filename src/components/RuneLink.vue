<template>
  <div class="rune-link" :class="{ active }">&nbsp;</div>
</template>

<script>
export default {
  props: {
    active: {
      type: Boolean,
      required: true
    }
  }
};
</script>

<style lang="scss">
.rune-link {
  display: block;
  width: 100px;
  height: 15px;
  background: rgba(128, 128, 128, 0.15);
  transition: background-color 0.5s ease-out;
  border-top: 1px solid rgba(255, 255, 255, 0.15);
  border-bottom: 1px solid rgba(255, 255, 255, 0.15);
  margin-top: auto;
  margin-bottom: auto;

  &.active {
    background: rgba(128, 128, 128, 0.5);
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-bottom: 1px solid rgba(255, 255, 255, 0.5);
  }
}
</style>
