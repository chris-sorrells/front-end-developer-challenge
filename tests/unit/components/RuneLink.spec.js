import { shallowMount } from "@vue/test-utils";
import RuneLink from "@/components/RuneLink.vue";

describe("RuneLink.vue", () => {
  it("renders when active", () => {
    const wrapper = shallowMount(RuneLink, { propsData: { active: true } });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders when not active", () => {
    const wrapper = shallowMount(RuneLink, { propsData: { active: false } });
    expect(wrapper.element).toMatchSnapshot();
  });
});
