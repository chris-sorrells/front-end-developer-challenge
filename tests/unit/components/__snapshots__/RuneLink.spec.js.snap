// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuneLink.vue renders when active 1`] = `
<div
  class="rune-link active"
>
   
</div>
`;

exports[`RuneLink.vue renders when not active 1`] = `
<div
  class="rune-link"
>
   
</div>
`;
